addons/calendar/static/src/js/base_calendar.js

```diff
@@ -187,52 +190,44 @@ var CalendarNotification = Notification.extend({
         },
     },
 
-    init: function(parent, title, text, eid) {
-        this._super(parent, title, text, true);
-        this.eid = eid;
+    init: function(parent, message) {
+        this._super(parent, message.title, message.message, true);
+        this.eid = message.event_id;
+        this.message = message;
     },
 });
 
-WebClient.include({
-    get_next_notif: function() {
+var calendarAlarmManager = WebWidget.extend({
+    init: function(parent, options) {
         var self = this;
-
-        this.rpc("/calendar/notify")
-        .done(function(result) {
-            _.each(result, function(res) {
-                setTimeout(function() {
-                    // If notification not already displayed, we create and display it (FIXME is this check usefull?)
-                    if(self.$(".eid_" + res.event_id).length === 0) {
-                        self.notification_manager.display(new CalendarNotification(self.notification_manager, res.title, res.message, res.event_id));
-                    }
-                }, res.timer * 1000);
-            });
-        })
-        .fail(function(err, ev) {
-            if(err.code === -32098) {
-                // Prevent the CrashManager to display an error
-                // in case of an xhr error not due to a server error
-                ev.preventDefault();
-            }
-        });
+        this._super(parent);
+        this.bus = bus.bus;
+        this.bus.last = 0;
+        this.bus.on("notification", this, this.on_notification);
     },
-    check_notifications: function() {
-        var self = this;
-        this.get_next_notif();
-        this.intervalNotif = setInterval(function() {
-            self.get_next_notif();
-        }, 5 * 60 * 1000);
+    get_notif_box: function(me) {
+        return $(me).closest(".ui-notify-message-style");
     },
+    on_notification: function(notification) {
+        var self = this;
+        var channel = notification[0];
+        var message = notification[1];
+
+        if (message.event_id){
+            setTimeout(function() {
+                self.getParent().notification_manager.display(new CalendarNotification(self.getParent().notification_manager, message));
+            }, message.timer * 1000);
```

Comment:
That is because now you created a CalendarAlarm widget so you will not get it in current element, may be you need to do self.getParent().$(...), but do not use global selector